$Id: trayer+dclock.combined_GtkTooltip_DClockUpdate.patch 1299 2008-03-17 02:12:14Z mehani $
Update Trayer's code to the new GtkTooltip feature instead of the GtkTooltips
formerly in use and deprecated as of GTK 2.12.
Update the clock using the specific functions as soon as the label is created to
avoid style inconsistencies.

Olivier Mehani <shtrom@ssji.net>
===
diff -urN trayer-1.0/dclock.c trayer-1.0-GtkTooltip/dclock.c
--- trayer-1.0/dclock.c	2008-03-17 12:54:58.000000000 +1100
+++ trayer-1.0-GtkTooltip/dclock.c	2008-03-17 13:01:06.000000000 +1100
@@ -22,7 +22,6 @@
     GtkWidget *eb;
     GtkWidget *main;
     GtkWidget *clockw;
-    GtkTooltips *tip;
     char *tfmt;
     char *cfmt;
     char *action;
@@ -69,7 +68,7 @@
 
 	strftime (output, sizeof(output), dc->tfmt, detail) ;
         if ((utf8 = g_locale_to_utf8(output, -1, NULL, NULL, NULL))) {
-            gtk_tooltips_set_tip(dc->tip, dc->main, utf8, NULL) ;
+	    gtk_widget_set_tooltip_text (dc->main, utf8);
             g_free(utf8);
         }
     }
@@ -126,15 +125,13 @@
         g_signal_connect (G_OBJECT (dc->main), "button_press_event",
               G_CALLBACK (clicked), (gpointer) dc);
     time(&now);
-    detail = localtime(&now);
-    strftime(output, sizeof(output), dc->cfmt, detail) ;
     dc->clockw = gtk_label_new(output);
+    clock_update(dc);
     gtk_misc_set_alignment(GTK_MISC(dc->clockw), 0.5, 0.5);
     gtk_misc_set_padding(GTK_MISC(dc->clockw), 4, 0);
     //gtk_widget_show(dc->clockw);
     gtk_container_add(GTK_CONTAINER(dc->main), dc->clockw);
     gtk_widget_show_all(dc->main);
-    dc->tip = gtk_tooltips_new();
     dc->timer = g_timeout_add(1000, (GSourceFunc) clock_update, (gpointer)dc);
     gtk_container_add(GTK_CONTAINER(p->pwid), dc->main);
     RET(1);
diff -urN trayer-1.0/systray/fixedtip.c trayer-1.0-GtkTooltip/systray/fixedtip.c
--- trayer-1.0/systray/fixedtip.c	2004-05-24 04:57:04.000000000 +1000
+++ trayer-1.0-GtkTooltip/systray/fixedtip.c	2008-03-17 13:03:23.000000000 +1100
@@ -36,17 +36,6 @@
   return FALSE;
 }
 
-static gboolean
-expose_handler (GtkTooltips *tooltips)
-{
-  gtk_paint_flat_box (tip->style, tip->window,
-                      GTK_STATE_NORMAL, GTK_SHADOW_OUT, 
-                      NULL, tip, "tooltip",
-                      0, 0, -1, -1);
-
-  return FALSE;
-}
-
 void
 fixed_tip_show (int screen_number,
                 int root_x, int root_y,
@@ -81,11 +70,6 @@
       gtk_widget_set_name (tip, "gtk-tooltips");
       gtk_container_set_border_width (GTK_CONTAINER (tip), 4);
 
-      g_signal_connect (G_OBJECT (tip), 
-            "expose_event",
-            G_CALLBACK (expose_handler),
-            NULL);
-   
       gtk_widget_add_events (tip, GDK_BUTTON_PRESS_MASK);
       
       g_signal_connect (G_OBJECT (tip), 
